src/FSharp.Control.TaskSeq/TaskSeq.fsi

```diff
@@ -154,414 +175,872 @@ module TaskSeq =
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
+    /// <exception cref="T:ArgumentException">Thrown when the sequence has insufficient length or <paramref name="index" /> is negative.</exception>
     val item: index: int -> source: taskSeq<'T> -> Task<'T>
 
     /// <summary>
     /// Returns the only element of the task sequence, or <see cref="None" /> if the sequence is empty of
     /// contains more than one element.
     /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The only element of the singleton task sequence, or <see cref="None" />.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
     val tryExactlyOne: source: taskSeq<'T> -> Task<'T option>
 
     /// <summary>
     /// Returns the only element of the task sequence.
     /// </summary>
-    /// <exception cref="ArgumentException">Thrown when the input sequence does not contain precisely one element.</exception>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The only element of the singleton task sequence, or <see cref="None" />.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
+    /// <exception cref="T:ArgumentException">Thrown when the input task sequence does not contain precisely one element.</exception>
     val exactlyOne: source: taskSeq<'T> -> Task<'T>
 
     /// <summary>
     /// Applies the given function <paramref name="chooser" /> to each element of the task sequence. Returns
-    /// a sequence comprised of the results "x" for each element where
-    /// the function returns <c>Some(x)</c>.
+    /// a sequence comprised of the results where the function returns <see cref="Some(x)" />.
     /// If <paramref name="chooser" /> is asynchronous, consider using <see cref="TaskSeq.chooseAsync" />.
     /// </summary>
+    ///
+    /// <param name="chooser">A function to transform items of type <typeref name="'T" /> into options of type <typeref name="'U" />.</param>
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
     val choose: chooser: ('T -> 'U option) -> source: taskSeq<'T> -> taskSeq<'U>
 
     /// <summary>
-    /// Applies the given asynchronous function <paramref name="chooser" /> to each element of the task sequence. Returns
-    /// a sequence comprised of the results "x" for each element where
-    /// the function returns <see cref="Some(x)" />.
-    /// If <paramref name="chooser" /> does not need to be asynchronous, consider using <see cref="TaskSeq.choose" />.
+    /// Applies the given asynchronous function <paramref name="chooser" /> to each element of the task sequence.
```

Comment:
Stopping here, the fact it takes ages to review reflects the insane amount of work this must have been to do!

was focused on individual wordings tat the start but have been scanning for a while. If you do find it within you to do the drudge work of addressing my suggestions, I promise to do a thorough re-review to trap inconsistencies on the ones I've already traversed, along with reading the rest thoroughly.

Reply:
@bartelink it is much appreciated! I think I've addressed all your comments so far. I appreciate the improved wording. Keep in mind that I seek parity with `seq` also in the use of wording (at least where it makes sense), to give people an easy adoption experience.

Reply:
Please do further comments in #136, which is where the changes will go.